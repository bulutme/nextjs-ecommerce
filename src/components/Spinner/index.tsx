import { FC } from "react";
import styled, { keyframes } from "styled-components";

const rotate = keyframes`
  to {
    transform: rotate(360deg);
  }
`;

const SpinnerStyled = styled.div`
  border: 4px solid rgba(0, 0, 0, 0.1);
  border-top: 4px solid ${({ theme }) => theme.colors.primary};
  border-radius: 50%;
  width: 64px;
  height: 64px;
  animation: ${rotate} 1s linear infinite;
`;

const Spinner: FC = () => {
  return <SpinnerStyled />;
};

export default Spinner;
