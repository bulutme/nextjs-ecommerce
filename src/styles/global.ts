"use client";

import { createGlobalStyle } from "styled-components";

const GlobalStyles = createGlobalStyle`
  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }

  html {
    font-size: 16px;
  }

  html, body, #__next {
    height: 100%;
  }

  a {
    text-decoration: none;
    width: fit-content;
  }

  ul {
    list-style-type: none
  }
`;

export default GlobalStyles;
