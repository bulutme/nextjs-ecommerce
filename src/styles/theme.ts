export const theme = {
  colors: {
    black: "#000",
    white: "#fff",
    gray: "#808080",
    grayLight: "#d3d3d3",
    border: "#e2e8f0",
    primary: "#3bb77e",
    primaryDark: "#2b8a5e",
    error: "#dc2626",
    info: "#0891b2",
    success: "#16a34a",
  },
  screens: {
    "2xs": "400px",
    xs: "480px",
    sm: "576px",
    md: "768px",
    lg: "992px",
    xl: "1200px",
    "2xl": "1440px",
    "3xl": "1560px",
  },
  fontWeight: {
    normal: 400,
    medium: 500,
    semibold: 600,
    bold: 700,
  },
};
