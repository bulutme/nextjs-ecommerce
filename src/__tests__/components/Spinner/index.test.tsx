import React from "react";
import { render } from "@testing-library/react";
import Spinner from "@/components/Spinner";
import { ThemeProvider } from "styled-components";
import { theme } from "@/styles/theme";

test("renders Spinner correctly", () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <Spinner />
    </ThemeProvider>
  );

  expect(container.firstChild).toMatchSnapshot();
});
