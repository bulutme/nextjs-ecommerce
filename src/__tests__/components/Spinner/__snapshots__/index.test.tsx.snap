// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Spinner correctly 1`] = `
<div
  class="sc-ecd6f36e-0 eVitae"
/>
`;
